tests/terraform/graph/checks/test_yaml_connected_nodes.py

```diff
@@ -48,17 +50,45 @@ def test_S3BucketLogging_connected_node(self):
         assert report.passed_checks[1].connected_node['resource'] == 'aws_s3_bucket_logging.example'
         assert report.passed_checks[1].connected_node['file_line_range'] == [14, 19]
 
-    def get_report(self, dir_name):
+    def test_correct_connected_node_per_check(self):
+        failed_check_id_with_connected_node = 'CKV_AWS_144'
+        pass_check_id_with_connected_node = 'CKV_AWS_145'
+        pass_check_id_without_connected_node = 'CKV_AWS_20'
+        check_ids = [failed_check_id_with_connected_node, pass_check_id_with_connected_node, pass_check_id_without_connected_node]
+
+        for graph_framework in ["NETWORKX", "RUSTWORKX"]:
```

Comment:
we have this parametrization in other places in the code, you can also use `pytest.mark.parametrize`